src/compiler/program.ts

```diff
@@ -908,6 +908,13 @@ namespace ts {
 
         function getSemanticDiagnosticsForFileNoCache(sourceFile: SourceFile, cancellationToken: CancellationToken): Diagnostic[] {
             return runWithCancellationToken(() => {
+                // If skipLibCheck is enabled, skip reporting errors if file is a declaration file.
+                // If skipDefaultLibCheck is enabled, skip reporting errors if file contains a
+                // '/// <reference no-default-lib="true"/>' directive.

```

Comment:
`no-default-lib` is just a triple-slash directive. it can exist any where. 
We have two flags, for historic reasons:
1. skipDefaultLibCheck, only skips the checking of files that have `/// <reference no-default-lib="true"/>` in them. This flag is now deprecated, but we still honor it.
2. skipLibCheck, skips checking of all .d.ts files